src/Router.sol

```diff
@@ -193,195 +209,173 @@ contract Router is Ownable, ReentrancyGuard, IRouter {
     }
 
-    /// @notice Builds a new transfer parameter object
-    function buildTransferParams(
+    /// @notice Builds swap request parameters based on the provided details
+    /// @param token The address of the token to be swapped
+    /// @param amount The amount of tokens to be swapped
+    /// @param verificationFeeAmount The verification fee amount
+    /// @param solverFeeAmount The solver fee amount
+    /// @param dstChainId The destination chain ID
+    /// @param recipient The address that will receive the tokens
+    /// @param nonce A unique nonce for the request
+    /// @return swapRequestParams A SwapRequestParameters struct containing the transfer parameters.
+    function buildSwapRequestParameters(
         address token,
         uint256 amount,
-        uint256 swapFeeAmount,
+        uint256 verificationFeeAmount,
         uint256 solverFeeAmount,
         uint256 dstChainId,
         address recipient,
         uint256 nonce
-    ) public view returns (TransferParams memory params) {
-        params = TransferParams({
+    ) public view returns (SwapRequestParameters memory swapRequestParams) {
+        swapRequestParams = SwapRequestParameters({
             sender: msg.sender,
             recipient: recipient,
             token: token,
             amount: amount,
-            srcChainId: thisChainId,
+            srcChainId: getChainID(),
             dstChainId: dstChainId,
-            swapFee: swapFeeAmount,
+            verificationFee: verificationFeeAmount,
             solverFee: solverFeeAmount,
             nonce: nonce,
             executed: false
         });
     }
 
-    /// @notice Stores a transfer request and marks as unfulfilled
-    function storeTransferRequest(bytes32 requestId, TransferParams memory params) internal {
-        transferParameters[requestId] = params;
-        unfulfilledSolverRefunds.add(requestId);
-    }
-
-    /// @notice Compares two transfer parameter structs
-    function isEqual(TransferParams memory a, TransferParams memory b) internal pure returns (bool) {
-        return a.sender == b.sender && a.recipient == b.recipient && a.token == b.token && a.amount == b.amount
-            && a.srcChainId == b.srcChainId && a.dstChainId == b.dstChainId && a.swapFee == b.swapFee
-            && a.solverFee == b.solverFee && a.nonce == b.nonce && a.executed == b.executed;
-    }
-
-    /// @notice Computes the swap fee in underlying token units
-    function getSwapFeeAmount(uint256 amount) public view returns (uint256) {
-        if (swapFeeBps == 0) return 0;
-        return (amount * swapFeeBps) / BPS_DIVISOR;
+    /// @notice Calculates the verification fee amount based on total fees
+    /// @param totalFees The total fees for which the verification fee is to be calculated
+    /// @return The calculated verification fee amount
+    function getVerificationFeeAmount(uint256 totalFees) public view returns (uint256) {
```

Comment:
it's still slightly surprising to me that we calculate the verification fee off the total fees and not the total amount. This suggests that a $1MM tx with a $1 fee will pay the same dcipher fee as a $10 tx with a $1 fee.
I'm not _against_ this, but there are cryptoeconomic implications for securing the network that we should discuss more indepth